internal/framework/search.go

```diff
@@ -0,0 +1,152 @@
+package framework
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/saucelabs/saucectl/internal/node"
+)
+
+// FrameworkUnavailableError is an error type returned if the requested framework version is unavailable.
+type FrameworkUnavailableError struct {
+	Name    string
+	Version string
+}
+
+// Error returns the error string
+func (e *FrameworkUnavailableError) Error() string {
+	s := fmt.Sprintf("version %s for %s is not available", e.Version, e.Name)
+	return s
+}
+
+// Misc errors
+var (
+	ErrServerError      = errors.New("unable to check framework version availability")
+	ErrVersionUndefined = errors.New("framework version is not defined")
+)
+
+// MetadataSearchStrategy is a generic strategy for determining if the requested framework version is supported
+type MetadataSearchStrategy interface {
+	Find(ctx context.Context, svc MetadataService, frameworkName string, searchValue string) (Metadata, error)
+}
+
+// ExactStrategy searches for the metadata of a framework by its exact version
+type ExactStrategy struct {
+}
+
+// PackageStrategy searches for metadata of a framework from a package.json file. If the requested version is a range (e.g. ~9.0.0), it matches the most recent version that satifies the constraint.
+type PackageStrategy struct {
+	packageJsonFilePath string
+}
+
+func (s ExactStrategy) Find(ctx context.Context, svc MetadataService, frameworkName string, frameworkVersion string) (Metadata, error) {
+	if frameworkVersion == "" {
+		return Metadata{}, ErrVersionUndefined
+	}
+
+	m, err := svc.Search(ctx, SearchOptions{
+		Name:             frameworkName,
+		FrameworkVersion: frameworkVersion,
+	})
+
+	if err != nil && strings.Contains(err.Error(), "Bad Request: unsupported version") {
+		return Metadata{}, &FrameworkUnavailableError{
+			Name:    frameworkName,
+			Version: frameworkVersion,
+		}
+	}
+
+	if err != nil {
+		return Metadata{}, ErrServerError
+	}
+
+	return m, nil
+}
+
+func toNpmPackageName(frameworkName string) string {
+	mapping := map[string]string{
+		"cypress":    "cypress",
+		"playwright": "@playwright/test",
```

Comment:
Shouldn't we rely on playwright package more that `@playwright/test` ?

Can't `@playwright/test` be installed globally on devs machine and simply rely on `playwright` ?

Reply:
I don't think so? Currently both installation methods described by the playwright docs results in `@playwright/test` being installed, not `playwright`.

Reply:
Probably the doc evolved :) I recall that it was a 2-step  thing. I'm glad it's not required anymore !